### PR TITLE
feat : added context and reducer for like , history and watch later

### PR DESCRIPTION
**Pull request by Omkar-Chaskar**
Added functional buttons -
Like button / Unlike button
Add to Watch Later later / remove from watch later
History / remove from history button

with Context provider and Reducers

### CODE REVIEW

File: src/component/home.js [hunk 1]
```diff
@@ -22,7 +26,26 @@ export default function Home() {
               <li className="card-vertical" key={video._id}>
                 <div className="card__primary-action card__primary-action-column card__primary-action-vertical">
                   <span className="material-icons-outlined badge-up-right-corner card-badge card-badge-vertical">
-                    <AiTwotoneLike />
+                    {likeState.likeVideos &&
+                    likeState.likeVideos.some(
+                      (item) => video._id === item._id
+                    ) ? (
```
Comment (JantuDeb) [inline src/component/home.js:32]: create a variable 

```js 
const isLiked = likeState.likeVideos.some(
                      (item) => video._id === item._id
                    ) 
```

File: src/component/likeVideo.js [hunk 2]
```diff
@@ -1,189 +1,83 @@
...
+          {likeState.likeVideos.map((video) => {
+            return (
+              <li className="card-vertical" key={video._id}>
                 <div className="card__primary-action card__primary-action-column card__primary-action-vertical">
                   <span className="material-icons-outlined badge-up-right-corner card-badge card-badge-vertical">
-                    <AiTwotoneLike color="red"/>
+                    {likeState.likeVideos &&
+                    likeState.likeVideos.some(
+                      (item) => video._id === item._id
+                    ) ? (
```
Comment (JantuDeb) [inline src/component/likeVideo.js:30]: you can use the save `isLiked` variable here also

File: src/component/singleVideo.js [hunk 3]
```diff
@@ -36,12 +42,58 @@ const SingleVideo = () => {
...
               <p>{video.creator}</p>
               <div className="btn-container">
-                <button className="btn-like button button-danger p1 bold">
-                  Like
-                </button>
-                <button className="btn-watch-later button button-primary p1 bold">
-                  Add to Watch Later
-                </button>
+                {likeState.likeVideos &&
+                likeState.likeVideos.some((item) => video._id === item._id) ? (
```
Comment (JantuDeb) [inline src/component/singleVideo.js:46]: same here

File: src/component/singleVideo.js [hunk 4]
```diff
@@ -36,12 +42,58 @@ const SingleVideo = () => {
...
+                      likeDispatch({ type: "ADD_TO_LIKE", payload: video })
+                    }
+                  >
+                    Like
+                  </button>
+                )}
+
+                {watchLaterState.watchLater &&
+                watchLaterState.watchLater.some(
+                  (item) => video._id === item._id
```
Comment (JantuDeb) [inline src/component/singleVideo.js:71]: You can do the same for watch later

**Review by JantuDeb — APPROVED** [review]
Overall looks fine, added few suggestion

**Review by Rohitprasad83 — APPROVED** [review]
Code is readable and UI also looks good.

File: src/component/home.js (hunk 1 above)
Comment (Omkar-Chaskar) [reply src/component/home.js:32]: Thanks for suggestion . I will do it in next PR.

**Pull request merged**